### PR TITLE
Test Explorer - Part 2

### PR DESCRIPTION
**Pull request by xisui-MSFT**
#1682 
- Added the ability to show test messages.
- Support for Catch test framework is removed:
  - Catch has been replaced by Catch2 since about 5 years ago.
  - Catch is not related to CMake/CTest so shouldn't be supported by this extension in the first place.
  - There is an extension, "C++ TestMate", providing excellent Catch2 support with the Test API.

Demo:
TestExplorer

### CODE REVIEW

File: src/ctest.ts
Comment (benmcmorran) [inline src/ctest.ts:300]: For my own understanding, are we only logging errors that don't have a URI and range because errors that have a URI and range will be directly displayed in the VS Code UI?

**Review by benmcmorran — APPROVED** [review]
Same as part 1, please update CHANGELOG.md with any breaking changes and new features in this PR.

**Comment by xisui-MSFT** [comment]
> Same as part 1, please update CHANGELOG.md with any breaking changes and new features in this PR.

I'll change CHANGELOG in part 3 for convenience.

**Comment by benmcmorran** [comment]
@xisui-MSFT looks like there's one small merge conflict in `ctest.ts` after part 1 merged. This should be good to merge once that's fixed.

**Comment by xisui-MSFT** [comment]
> @xisui-MSFT looks like there's one small merge conflict in `ctest.ts` after part 1 merged. This should be good to merge once that's fixed.

Done. The diff on this website looks wrong for some reason.

**Review by benmcmorran — APPROVED** [review]

**Pull request merged**
